src/program/render.rs

```diff
@@ -0,0 +1,225 @@
+use crate::infobar;
+use crate::program::{make_dst, Program};
+use sdl2::image::LoadTexture;
+use sdl2::pixels::Color;
+use sdl2::rect::Rect;
+
+const PADDING: i32 = 30;
+const HALF_PAD: i32 = 15;
+
+impl<'a> Program<'a> {
+    /// render loads the image at the path in the images path vector located at the index and
+    /// renders to screen
```

Comment:
Maybe changing the `render` function to `render_screen` or something that indicates that it is essentially doing everything to update the screen (clear canvas, render image (same or new depending on ui state changes), and present changes to user)